package com.back.boundedContext.post.domain;

import com.back.global.jpa.entity.BaseIdAndTime;
import com.back.shared.post.dto.PostCommentDto;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "POST_POST_COMMENT")
@NoArgsConstructor
@Getter
public class PostComment extends BaseIdAndTime {

	@ManyToOne(fetch = FetchType.LAZY)
	private Post post;
	@ManyToOne(fetch = FetchType.LAZY)
	private PostMember author;
	@Column(columnDefinition = "TEXT")
	private String content;

	public PostComment(Post post, PostMember author, String content) {
		this.post = post;
		this.author = author;
		this.content = content;
	}

	public PostCommentDto toDto() {
		return new PostCommentDto(
			getId(),
			getCreateTime(),
			getModifyTime(),
			post.getId(),
			author.getId(),
			author.getNickName(),
			content
		);
	}


}
